src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {
+    View,
+    Text,
+    TouchableOpacity,
+    ActivityIndicator,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import ONYXKEYS from '../../../ONYXKEYS';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+
+const propTypes = {
+    // Callback to inform parent modal of success
+    onStepComplete: PropTypes.func.isRequired,
+
+    /* Onyx Props */
+
+    // Holds data related to IOU view state, rather than the underlying IOU data.
+    appStateIOU: PropTypes.shape({
```

Comment:
Still not super clear to me why we need to have this convention/distinction. Why should the IOU data not be an object with a `loading` prop? And if we are making an `appStatIOU` object could we just as easily have an `isIOULoading: Boolean` Onyx key?

Reply:
I don't really have a preference, but further discussion can be found on [this thread](https://github.com/Expensify/Expensify.cash/pull/1561#discussion_r588406426) led to this change. Would you mind commenting there to continue the discussion?